File: frontend/src/lib/services/resource.ts
import { apiFetch } from './api';

const API_URL = '/api/v1';

export interface WorkloadEntry {
    user_id: number;
    full_name: string;
    email: string;
    role: string;
    date: string;
    task_count: number;
    total_hours: number;
    load_percentage: number;
}

export interface ResourceWorkload {
    user_id: number;
    full_name: string;
    email: string;
    role: string;
    entries: WorkloadEntry[];
}

export interface WorkloadOverview {
    start_date: string;
    end_date: string;
    users: ResourceWorkload[];
}

export const resourceService = {
    async getWorkload(startDate: string, endDate: string): Promise<WorkloadOverview> {
        const res = await apiFetch(`${API_URL}/resources/workload?start_date=${startDate}&end_date=${endDate}`);
        if (!res.ok) throw new Error('Failed to load workload data');
        return res.json();
    }
};

File: backend/internal/domain/entity/resource.go
package entity

// ResourceWorkloadEntry aggregates workload data for a user on a single day
type ResourceWorkloadEntry struct {
	UserID         int     `json:"user_id"`
	FullName       string  `json:"full_name"`
	Email          string  `json:"email"`
	Role           string  `json:"role"`
	Date           string  `json:"date"` // YYYY-MM-DD
	TaskCount      int     `json:"task_count"`
	TotalHours     float64 `json:"total_hours"`     // estimated hours per day
	LoadPercentage float64 `json:"load_percentage"` // e.g., (TotalHours / 8.0) * 100
}

// ResourceWorkload groups workload entries by user
type ResourceWorkload struct {
	UserID   int                     `json:"user_id"`
	FullName string                  `json:"full_name"`
	Email    string                  `json:"email"`
	Role     string                  `json:"role"`
	Entries  []ResourceWorkloadEntry `json:"entries"`
}

// WorkloadOverview is the top-level response
type WorkloadOverview struct {
	StartDate string             `json:"start_date"`
	EndDate   string             `json:"end_date"`
	Users     []ResourceWorkload `json:"users"`
}
